lib/src/vector_math/vector2.dart

```diff
@@ -395,21 +377,21 @@ class Vector2 implements Vector {
   /// Copy this into [arg]. Returns [arg].
   Vector2 copyInto(Vector2 arg) {
     final argStorage = arg._v2storage;
-    argStorage[1] = _v2storage[1];
     argStorage[0] = _v2storage[0];
+    argStorage[1] = _v2storage[1];
     return arg;
   }
 
   /// Copies this into [array] starting at [offset].
   void copyIntoArray(List<double> array, [int offset = 0]) {
-    array[offset + 1] = _v2storage[1];
     array[offset + 0] = _v2storage[0];
+    array[offset + 1] = _v2storage[1];
```

Comment:
The ordering here of accessing the higher index first is intentional, it guarantees that bounds checking is only done once.

Reply:
This applies to a bunch of places in this PR.

Reply:
Ah, that makes sense, let me fix that.

Reply:
Done!